java/ingest/ingest-batcher-submitter/src/main/java/sleeper/ingest/batcher/submitter/IngestBatcherSubmitterLambda.java

```diff
@@ -67,22 +73,58 @@ public Void handleRequest(SQSEvent input, Context context) {
     }
 
     public void handleMessage(String json, Instant receivedTime) {
-        FileIngestRequest request;
+        List<FileIngestRequest> requests;
         try {
-            request = FileIngestRequestSerDe.fromJson(json, receivedTime);
+            requests = FileIngestRequestSerDe.fromJson(json, receivedTime);
         } catch (RuntimeException e) {
             LOGGER.warn("Received invalid ingest request: {}", json, e);
             return;
         }
-
         // Table properties are needed to set the expiry time on DynamoDB records in the store.
         // To avoid that failing, we can discard the message here if the table does not exist.
-        if (tablePropertiesProvider.getTablePropertiesIfExists(request.getTableName())
-                .isEmpty()) {
+        if (requests.size() > 0 &&
+                tablePropertiesProvider.getTablePropertiesIfExists(requests.get(0).getTableName()).isEmpty()) {
             LOGGER.warn("Table does not exist for ingest request: {}", json);
             return;
         }
-        LOGGER.info("Adding {} to store", request.getFile());
-        store.addFile(request);
+        requests.forEach(request -> storeFiles(request, receivedTime));
+    }
+
+    private void storeFiles(FileIngestRequest request, Instant receivedTime) {
+        String bucketName = getBucketName(request);
+        String filePath = getFilePath(request);
+        ObjectListing result = s3Client.listObjects(bucketName, filePath);
+        result.getObjectSummaries().stream()
+                .map(summary -> toRequest(summary, bucketName, request.getTableName(), receivedTime))
+                .forEach(store::addFile);
+        if (result.getObjectSummaries() != null && result.getObjectSummaries().stream()
+                .noneMatch(summary -> summary.getKey().equals(getFilePath(request)))) {
+            result.getCommonPrefixes().forEach(directory ->
+                    streamFilesInDirectory(s3Client, bucketName, directory)
+                            .map(summary -> toRequest(summary, bucketName, request.getTableName(), receivedTime))
+                            .forEach(store::addFile));
```

Comment:
This code is never hit by any of the tests. It looks like getCommonPrefixes is only used if you specify a delimiter in the request.